Panic: runtime error when running terragrunt validate-all

After adding `terraform_version_constraint` and `terragrunt_version_constraint` to our root `terragrunt.hcl` file, we are getting a Panic during the Terragrunt version check.

```terraform
terraform_version_constraint  = ">= 0.12.20"
terragrunt_version_constraint = ">= 0.23.20"
```

Terraform version: 0.12.26
Terragrunt version: 0.23.20
OS: Docker python:3.8.1-slim

Command:
`terragrunt validate-all -no-color`

backtrace:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x6e6036]

goroutine 46 [running]:
github.com/hashicorp/go-version.(*Version).Prerelease(...)
        /go/pkg/mod/github.com/hashicorp/go-version@v1.2.0/version.go:322
github.com/hashicorp/go-version.prereleaseCheck(0x0, 0xc0003f2500, 0x1)
        /go/pkg/mod/github.com/hashicorp/go-version@v1.2.0/constraint.go:118 +0x26
github.com/hashicorp/go-version.constraintGreaterThanEqual(0x0, 0xc0003f2500, 0xc0003fa140)
        /go/pkg/mod/github.com/hashicorp/go-version@v1.2.0/constraint.go:158 +0x35
github.com/hashicorp/go-version.(*Constraint).Check(...)
        /go/pkg/mod/github.com/hashicorp/go-version@v1.2.0/constraint.go:92
github.com/hashicorp/go-version.Constraints.Check(0xc0000c40e0, 0x1, 0x1, 0x0, 0x1)
        /go/pkg/mod/github.com/hashicorp/go-version@v1.2.0/constraint.go:72 +0x5c
github.com/gruntwork-io/terragrunt/cli.checkTerragruntVersionMeetsConstraint(0x0, 0xc0004e45c0, 0xa, 0x0, 0x0)
        /go/src/github.com/gruntwork-io/terragrunt/cli/version_check.go:58 +0x90
github.com/gruntwork-io/terragrunt/cli.CheckTerragruntVersion(...)
        /go/src/github.com/gruntwork-io/terragrunt/cli/version_check.go:48
github.com/gruntwork-io/terragrunt/cli.checkVersionConstraints(0xc000568780, 0x2, 0x2)
        /go/src/github.com/gruntwork-io/terragrunt/cli/cli_app.go:404 +0x180
github.com/gruntwork-io/terragrunt/cli.RunTerragrunt(0xc000568780, 0x13db063, 0x15)
        /go/src/github.com/gruntwork-io/terragrunt/cli/cli_app.go:278 +0xed
github.com/gruntwork-io/terragrunt/configstack.(*runningModule).runNow(0xc00010ca50, 0xc000429f18, 0x0)
        /go/src/github.com/gruntwork-io/terragrunt/configstack/running_module.go:243 +0x16f
github.com/gruntwork-io/terragrunt/configstack.(*runningModule).runModuleWhenReady(0xc00010ca50, 0xc000493020)
        /go/src/github.com/gruntwork-io/terragrunt/configstack/running_module.go:206 +0xd0
github.com/gruntwork-io/terragrunt/configstack.runModules.func1(0xc0000c0a00, 0xc000493020, 0xc00010ca50)
        /go/src/github.com/gruntwork-io/terragrunt/configstack/running_module.go:172 +0x65
created by github.com/gruntwork-io/terragrunt/configstack.runModules
        /go/src/github.com/gruntwork-io/terragrunt/configstack/running_module.go:170 +0x112
```

--- a/options/options.go
+++ b/options/options.go
@@ -222,6 +222,7 @@ func (terragruntOptions *TerragruntOptions) Clone(terragruntConfigPath string) *
 		TerraformPath:               terragruntOptions.TerraformPath,
 		TerraformCommand:            terragruntOptions.TerraformCommand,
 		TerraformVersion:            terragruntOptions.TerraformVersion,
+		TerragruntVersion:           terragruntOptions.TerragruntVersion,
 		AutoInit:                    terragruntOptions.AutoInit,
 		NonInteractive:              terragruntOptions.NonInteractive,
 		TerraformCliArgs:            util.CloneStringList(terragruntOptions.TerraformCliArgs),

--- a/test/fixture-version-check/a/main.tf
+++ b/test/fixture-version-check/a/main.tf
@@ -1,0 +1,3 @@
+output "foo" {
+  value = "Hello, World"
+}

--- a/test/fixture-version-check/a/terragrunt.hcl
+++ b/test/fixture-version-check/a/terragrunt.hcl
@@ -1,0 +1,3 @@
+include {
+  path = find_in_parent_folders()
+}

--- a/test/fixture-version-check/b/main.tf
+++ b/test/fixture-version-check/b/main.tf
@@ -1,0 +1,3 @@
+output "foo" {
+  value = "Hello, World"
+}

--- a/test/fixture-version-check/b/terragrunt.hcl
+++ b/test/fixture-version-check/b/terragrunt.hcl
@@ -1,0 +1,3 @@
+include {
+  path = find_in_parent_folders()
+}

--- a/test/fixture-version-check/terragrunt.hcl
+++ b/test/fixture-version-check/terragrunt.hcl
@@ -1,0 +1,2 @@
+terraform_version_constraint  = ">= 0.12.20"
+terragrunt_version_constraint = ">= 0.23.20"

--- a/test/integration_test.go
+++ b/test/integration_test.go
@@ -2875,6 +2875,19 @@ func TestTerragruntRemoteStateCodegenDoesNotGenerateWithSkip(t *testing.T) {
 	assert.False(t, fileIsInFolder(t, "foo.tfstate", generateTestCase))
 }
 
+func TestTerragruntValidateAllWithVersionChecks(t *testing.T) {
+	t.Parallel()
+
+	tmpEnvPath := copyEnvironment(t, "fixture-version-check")
+
+	stdout := bytes.Buffer{}
+	stderr := bytes.Buffer{}
+	err := runTerragruntVersionCommand(t, "v0.23.21", fmt.Sprintf("terragrunt validate-all --terragrunt-non-interactive --terragrunt-working-dir %s", tmpEnvPath), &stdout, &stderr)
+	logBufferContentsLineByLine(t, stdout, "stdout")
+	logBufferContentsLineByLine(t, stderr, "stderr")
+	require.NoError(t, err)
+}
+
 func TestTerragruntVersionConstraints(t *testing.T) {
 	testCases := []struct {
 		name                 string

